packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -6,6 +6,70 @@ import { Construct } from 'constructs';
 import { CfnJobDefinition } from './batch.generated';
 import { JobDefinitionImageConfig } from './job-definition-image-config';
 
+/**
+ * The log driver to use for the container.
+ */
+export enum LogDriver {
+  /**
+   * Specifies the Amazon CloudWatch Logs logging driver.
+   */
+  AWSLOGS = 'awslogs',
+
+  /**
+   * Specifies the Fluentd logging driver.
+   */
+  FLUENTD = 'fluentd',
+
+  /**
+   * Specifies the Graylog Extended Format (GELF) logging driver.
+   */
+  GELF = 'gelf',
+
+  /**
+   * Specifies the journald logging driver.
+   */
+  JOURNALD = 'journald',
+
+  /**
+   * Specifies the JSON file logging driver.
+   */
+  JSON_FILE = 'json-file',
+
+  /**
+   * Specifies the Splunk logging driver.
+   */
+  SPLUNK = 'splunk',
+
+  /**
+   * Specifies the syslog logging driver.
+   */
+  SYSLOG = 'syslog'
+}
+
+/**
+ * Log configuration options to send to a custom log driver for the container.
+ */
+export interface LogConfiguration {
+  /**
+   * The log driver to use for the container.
+   */
+  readonly logDriver: LogDriver;
+
+  /**
+   * The configuration options to send to the log driver.
+   *
+   * @default - No configuration options are sent
+   */
+  readonly options?: any;
+
+  /**
+   * The secrets to pass to the log configuration.
+   *
+   * @default - No secrets are passed
+   */
+  readonly secrets?: CfnJobDefinition.SecretProperty[];
```

Comment:
We don't expose L1 resource properties on the L2 layer. These properties are auto-generated and in the future might contain properties we don't want to directly expose. 

If needed, we just duplicate the current interface so we have more control. 

In this case though, I see the supported configuration is either a secret from AWS secretes manager, or a parameter from AWS SSM.

> The secret to expose to the container. The supported values are either the full ARN of the AWS Secrets Manager secret or the full ARN of the parameter in the AWS Systems Manager Parameter Store.

So this should be modeled accordingly by accepting either a [`Secret`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-secretsmanager/lib/secret.ts) or a [`Parameter`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-ssm/lib/parameter.ts).

Something like this:

```ts
new batch.JobDefinition(this, 'JobDefinition', {
  container: {
    logConfiguration: {
      logDriver: LogDriver.SYSLOG,
      secrets: [{
        name: 'secret',
        value: batch.Secret.fromSecretsManager(..), // batch.Secret.fromParametersStore(...)
      }]
    }
  }
});
```

Reply:
Added a concise implementation for this. Let me know if that makes sense for this usecase.